metrics/cloudwatch/cloudwatch.go

```diff
@@ -174,13 +175,32 @@ func (cw *CloudWatch) Send() error {
 	})
 
 	cw.gauges.Reset().Walk(func(name string, lvs lv.LabelValues, values []float64) bool {
-		value := last(values)
-		datums = append(datums, &cloudwatch.MetricDatum{
+		datum := &cloudwatch.MetricDatum{
 			MetricName: aws.String(name),
 			Dimensions: makeDimensions(lvs...),
-			Value:      aws.Float64(value),
 			Timestamp:  aws.Time(now),
-		})
+		}
+
+		if len(values) == 0 {
```

Comment:
We could move this above the datum initialization to save an alloc, but not a big deal.